CFC_DataCollector/modeinfer/pipeline.py

```diff
@@ -122,13 +159,15 @@ def generateFeatureMatrixAndResultVectorStep(self):
 
       # This will crash the script because we will try to access a record that
       # doesn't exist.
-
       # So we limit the records to the size of the matrix that we have created
       for (i, section) in enumerate(self.confirmedSections.limit(featureMatrix.shape[0]).batch_size(300)):
-        self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
+        #self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
+        featureMatrix[i] = self.updateFeatureMatrixRowWithSection(featureMatrix, i, section)
```

Comment:
Why return the row instead of updating the feature matrix directly? I don't have a strong reason why we need to update as a side effect, but am just curious about the reason for the change...